Feature/round of frontend cleanup

xtine opened the pull request:
## Summary
This is a round of front end cleanup/refactoring on JavaScript initialization files and modules.
 
- Remove old references to intro tour feature
  - Remove old `tour__point` classes that contained tour tooltip content and `TOUR_PAGE` JS global variables
- Remove references to `cms_url` and `webAppUrl` now that we can directly reference relative paths
- Remove extraneous code in site nav
- Remove extra typeahead initialization in Django base template
- Remove `StickyBar` module that is not being used anymore (was previously used on election page before redesign)
- Hide glossary from being revealed when hidden (trying to overscroll)
- Remove unused `between-commitees.jinja` template. https://github.com/18F/fec-cms/issues/1632
- Renamed `landing.js` to `data-landing.js` to be more specific that it is JS for the landing page of the data section.
- Fix glossary close page jump bug: https://github.com/18F/fec-cms/issues/1407
- Add documentation to not-in-use JS module `reaction-box.js`

## Impacted areas of the application
List general components of the application that this PR will affect:
-  Removes stale code for site nav and typeahead, will need to double check that this does not cause any regression errors.
- General scrolling on site to prevent overscrolling.

## Screenshots
Fixes this overscrolling issue:
<img width="1434" alt="screen shot 2017-12-11 at 3 04 56 pm" src="https://user-images.githubusercontent.com/24054/33859248-3ffb72fe-de90-11e7-95a9-d40f9a7242f8.png">

jenniferthibault commented on fec/data/templates/landing.jinja:
```diff
@@ -327,7 +327,5 @@
 <script src="{{ asset_for_js('landing.js') }}"></script>
-<script>
-  TOUR_PAGE = 'campaign-finance-data';
 </script>
```
Was this closing `<script>` supposed to get deleted too?

xtine replied:
Oof, good catch! Should have removed closing tag too.

xtine commented:
@jenniferthibault:
- fixed election map and line charts not showing up, it was because I tweaked how scrolling worked (to prevent overscrolling).
- apparently that dev banner has only just shown up on data (previously openfec-web-app), and it was still linking to beta.fec.gov so I just removed it entirely.

ccostino commented:
I tested this locally as well and came across a couple of things so far:

In the About section of the top nav menu, when you click on Working with the FEC, you're taken to the bottom of the page, but the menu on the left stays stuck at the top of the screen, so if you scroll up, it looks like this:

<img width="1261" alt="screen shot 2017-12-13 at 12 59 27 pm" src="https://user-images.githubusercontent.com/121565/33954224-9f830f68-e005-11e7-94d6-9ba13722c5a6.png">

In the Legal section of the top nav menu, clicking on Court Cases takes you to that page and the menu stays in place when you click and scroll around the page - is this expected?

<img width="1256" alt="screen shot 2017-12-13 at 1 08 03 pm" src="https://user-images.githubusercontent.com/121565/33954590-b41b41b0-e006-11e7-8118-8e4bbeab96e2.png">

All that said, most of the code changes themselves are okay as far as I can tell now!

xtine commented:
@ccostino: thanks for bringing up those bugs! so I realized the way I was "fixing" overscrolling caused weirdness on any sticky elements, including those side nav bars. So I found another solution that will still successfully hide glossary from overscrolling but doesn't touch the overflow settings of `html`/`body` to not mess around with the scroll event that was causing the problems before.

ccostino commented:
Thanks, @xtine!  Things look okay to me now in poking around, but will do a final code review again too. :-)

jenniferthibault commented:
Ah, @xtine I did find a thing that removing the sticky component broke: the left-hand-side jump menu, as seen on https://www.fec.gov/help-candidates-and-committees/filing-reports/

<img width="1114" alt="screen shot 2017-12-13 at 8 12 22 pm" src="https://user-images.githubusercontent.com/11636908/33970629-07b3041c-e042-11e7-8d5a-f4e8ae245007.png">

xtine commented:
The left sticky component shouldn’t be affected with my latest change.
